tap_github/client.py

```diff
@@ -26,7 +26,7 @@ def http_headers(self) -> dict:
         """Return the http headers needed."""
         headers = {"Accept": "application/vnd.github.v3+json"}
         if "user_agent" in self.config:
-            headers["User-Agent"] = self.config.get("user_agent")
+            headers["User-Agent"] = self.config.get("user_agent")  # type: ignore
```

Comment:
Is there a way to make mypy aware of the jsonschema for the config?

Reply:
There may be a json schema manner that is better, but this at least is explicit. Unlike in other languages, ["cast()"](https://mypy.readthedocs.io/en/stable/casts.html) is purely a type hint - but it nonetheless makes it easy for the reader to understand the expected behavior.

```suggestion
            headers["User-Agent"] = cast(str, self.config.get("user_agent"))
```

Reply:
It does look better indeed. Committed below. Thanks for the suggestion!